Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -747,6 +753,12 @@ extension Purchases {
 
 public extension Purchases {
 
+    /// Parses a deep link URL to verify it's a RevenueCat web purchase redemption link
+    /// - Seealso: ``Purchases/redeemWebPurchase(_:)``
+    @objc static func parseAsWebPurchaseRedemption(_ url: URL) -> WebPurchaseRedemption? {
```

Comment:
this API kind of feels off to me, for two reasons: 
1) it feels like leaked abstractions, don't know, I just feel awkward having to ask an SDK to parse a url as a web purchase redemption. Shouldn't the SDK just be able to take the URL? 
2) it doesn't quite roll off the tongue, now does it?

Reply:
I mentioned this during CoreSDK weekly, I feel like we can keep the functionality by splitting the API into "is this link for a web purchase redemption" and "redeem", i.e.:

```swift
func isWebPurchaseRedemptionLink(_ url: URL) -> Bool
func redeemWebPurchase(_ webPurchaseRedemption: WebPurchaseRedemption) async -> WebPurchaseRedemptionResult {
```

Reply:
From rereading, I think your preferred option would be something like:
```
func isWebPurchaseRedemptionLink(_ url: URL) -> Bool
func redeemWebPurchase(_ url: URL) async -> WebPurchaseRedemptionResult {
```
Where the `WebPurchaseRedemptionResult` has an `invalidLink` kind of result right? We can chat today to make sure we're in the same page.